Gems/Terrain/Code/Source/TerrainRenderer/Passes/TerrainClipmapComputePass.cpp

```diff
@@ -9,24 +9,152 @@
 #include <Atom/RPI.Public/Image/AttachmentImagePool.h>
 #include <Atom/RPI.Public/Pass/PassUtils.h>
 #include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/RPIUtils.h>
 #include <Atom/RPI.Public/Scene.h>
 #include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Reflect/Shader/ShaderAsset.h>
 #include <TerrainRenderer/Passes/TerrainClipmapComputePass.h>
 #include <TerrainRenderer/TerrainFeatureProcessor.h>
 #include <TerrainRenderer/TerrainClipmapManager.h>
 #include <Atom/RHI/FrameGraphAttachmentInterface.h>
 #include <Atom/RHI/FrameGraphBuilder.h>
+#include <Atom/RHI/CommandList.h>
 
 namespace Terrain
 {
+    TerrainClipmapGenerationPass::TerrainClipmapGenerationPass(const AZ::RPI::PassDescriptor& descriptor)
+        : AZ::RPI::RenderPass(descriptor)
+        , m_passDescriptor(descriptor)
+    {
+        LoadShader();
+    }
+
+    void TerrainClipmapGenerationPass::SetupFrameGraphDependencies(AZ::RHI::FrameGraphInterface frameGraph)
+    {
+        AZ::RPI::RenderPass::SetupFrameGraphDependencies(frameGraph);
+        frameGraph.SetEstimatedItemCount(1);
+    }
+
+    void TerrainClipmapGenerationPass::CompileResources(const AZ::RHI::FrameGraphCompileContext& context)
+    {
+        if (m_shaderResourceGroup != nullptr)
+        {
+            BindPassSrg(context, m_shaderResourceGroup);
+            m_shaderResourceGroup->Compile();
+        }
+        if (m_drawSrg != nullptr)
+        {
+            BindSrg(m_drawSrg->GetRHIShaderResourceGroup());
+            m_drawSrg->Compile();
+        }
+    }
+
+    void TerrainClipmapGenerationPass::BuildCommandListInternal(const AZ::RHI::FrameGraphExecuteContext& context)
+    {
+        //! Skip invoking the compute shader.
+        if (m_dispatchItem.m_arguments.m_direct.m_totalNumberOfThreadsX == 0 ||
+            m_dispatchItem.m_arguments.m_direct.m_totalNumberOfThreadsY == 0)
+        {
+            return;
+        }
```

Comment:
You shouldn't be skipping the command list submission at this point if X or Y = 0. Instead you can override the pass's IsEnabled() function and check in there and return false if X or Y = 0. This way you will bypass the pass completely and not declare dependencies to the FrameGraph, which might cause errors if you do and then don't submit any dispatch item.